src/llamafactory/train/dpo/workflow.py

```diff
@@ -64,6 +66,11 @@ def run_dpo(
     # Update arguments
     training_args.remove_unused_columns = False  # important for multimodal and pairwise dataset
 
+    effective_token_num = 0.0
+    for data in dataset_module["train_dataset"]:
```

Comment:
It may hurt the training performance for a large dataset, consider using a special CLI argument to control this behavior.

Reply:
okay